### PR TITLE
Have a topic per lobby

### PR DESCRIPTION
**Pull request by erikdubbelboer**
Instead of always getting a list of peers and pushing a message to each of then, we Subscribe on a lobby topic as well so we can publish on that.

### CODE REVIEW

File: internal/signaling/peer.go [hunk 1]
```diff
@@ -403,6 +397,7 @@ func (p *Peer) HandleJoinPacket(ctx context.Context, packet JoinPacket) error {
 	}
 
 	p.Lobby = packet.Lobby
+	p.store.Subscribe(ctx, p.Game+p.Lobby, p.ForwardMessage)
```
Comment (koenbollen) [inline internal/signaling/peer.go:400]: I think it smart to move the `lobby, err := p.store.GetLobby(ctx, p.Game, packet.Lobby)` below the store.Subscribe() calls. That way we'll never miss a peer when to peers connect at the same time.

File: internal/signaling/peer.go [hunk 2]
```diff
@@ -247,6 +243,7 @@ func (p *Peer) HandleHelloPacket(ctx context.Context, packet HelloPacket) error
 		if inLobby {
 			logger.Info("peer rejoining lobby", zap.String("game", p.Game), zap.String("peer", p.ID), zap.String("lobby", p.Lobby))
 			p.Lobby = lobby
+			p.store.Subscribe(ctx, p.Game+p.Lobby, p.ForwardMessage)
 			p.store.Subscribe(ctx, p.Game+p.Lobby+p.ID, p.ForwardMessage)
```
Comment (koenbollen) [inline internal/signaling/peer.go:247]: Suggestion (non blocking): Is it worth making this one call? e.g. `Subscribe(context.Context, cb func(), topics ...string)`

File: internal/signaling/peer.go (hunk 1 above)
Comment (erikdubbelboer) [reply internal/signaling/peer.go:400]: Good one.

File: internal/signaling/peer.go (hunk 2 above)
Comment (erikdubbelboer) [reply internal/signaling/peer.go]: I have made this change now. The nice thing is that the Subscribe happens in a single mutex lock. And that we only have one goroutine waiting to unsubscribe. But I think the Subscribe call itself is slightly less readable.

**Review by koenbollen — APPROVED** [review]

**Pull request merged**